services/sensiolabs/sensiolabs.service.js

```diff
@@ -1,110 +1,119 @@
 'use strict'
 
-const LegacyService = require('../legacy-service')
-const { makeBadgeData: getBadgeData } = require('../../lib/badge-data')
+const Joi = require('joi')
+const BaseXmlService = require('../base-xml')
 const serverSecrets = require('../../lib/server-secrets')
+const { statusRegex } = require('./sensiolabs-helpers')
+
+const schema = Joi.object({
+  project: Joi.object({
+    'last-analysis': Joi.object({
+      status: Joi.string()
+        .regex(statusRegex)
+        .required(),
+      grade: Joi.string()
+        .regex(/platinum|gold|silver|bronze|none/)
+        .allow('', null),
+    }),
+  }).required(),
+}).required()
+
+module.exports = class Sensiolabs extends BaseXmlService {
+  static render({ status, grade }) {
+    if (status !== 'finished') {
+      return {
+        message: 'pending',
+        color: 'grey',
```

Comment:
I wasn't sure what color to use here (when an analysis is still running) since orange is already used to indicate a different status (completed analysis, bronze medal)

Reply:
Is there a reference that describes what the grades mean?

Reply:
Not that I found (although I didn't spent a ton of time looking)

Reply:
https://insight.symfony.com/what-we-analyse

<img width="610" alt="screen shot 2018-12-21 at 1 46 11 pm" src="https://user-images.githubusercontent.com/1487036/50358334-ecb9b380-0526-11e9-8a97-2922467b61a3.png">

For the grade badge, I'd probably go with the flow of the design, and assign a color scheme that matches the words. (Their colors aren't bad except the platinum which is pretty green.)

If we can get the raw counts, I'd suggest we offer a second format, which is a bit more in shields-style, showing how many failed critical, major, minor, and info checks there are. In that case, we can use these colors:

<img width="992" alt="screen shot 2018-12-21 at 1 49 44 pm" src="https://user-images.githubusercontent.com/1487036/50358435-4b7f2d00-0527-11e9-8ddb-ce0a24ddad43.png">

One or more failed critical is red, one or more failed major is orange, minor yellow, info yellowgreen or green, and all good in bright green.

Something like:

- ![](https://img.shields.io/badge/checks-1%20critical-red.svg)
- ![](https://img.shields.io/badge/checks-1%20major,%201%20info-orange.svg)
- ![](https://img.shields.io/badge/checks-success-brightgreen.svg)

It looks like this product is called SymfonyInsight; maybe we should update our badge name.

Reply:
I like that a lot!

> It looks like this product is called SymfonyInsight; maybe we should update our badge name.

I had the same thought. Is it possible to configure the route base to support both `sensiolabs` and `symfony` to maintain backwards compatibility for existing badges? I.e. could we do something like:

```js
static get route() {
    return {
      base: '(sensiolabs|symfony)/i',
      pattern: ':projectUuid',
    }
  }
```

If so should the examples all focus on the "new" path (`symfony`) and not mention the legacy path (`sensiolabs`)?

Reply:
My local testing seems to indicate that won't work 😄

Reply:
I added support for including the raw results, as well as providing the corresponding badge as suggested above. I still need to update the colors for the grade badge though